engines/ospray/OSPRayFrameBuffer.cpp

```diff
@@ -90,42 +89,49 @@ void OSPRayFrameBuffer::_recreate()
 {
     auto lock = getScopeLock();
 
-    if (_frameBuffer)
-    {
-        _unmapUnsafe();
-        ospRelease(_frameBuffer);
-    }
-
-    OSPFrameBufferFormat format;
-    switch (_frameBufferFormat)
-    {
-    case FrameBufferFormat::rgba_i8:
-        format = OSP_FB_RGBA8;
-        break;
-    case FrameBufferFormat::rgb_f32:
-        format = OSP_FB_RGBA32F;
-        break;
-    default:
-        format = OSP_FB_NONE;
-    }
+    _unmapUnsafe();
+    ospRelease(_frameBuffer);
 
     const osp::vec2i size = {int(_frameSize.x()), int(_frameSize.y())};
 
     size_t attributes = OSP_FB_COLOR | OSP_FB_DEPTH;
     if (_accumulation)
         attributes |= OSP_FB_ACCUM | OSP_FB_VARIANCE;
 
-    _frameBuffer = ospNewFrameBuffer(size, format, attributes);
+    _frameBuffer = ospNewFrameBuffer(size, _ospFrameBufferFormat(), attributes);
     if (_pixelOp)
         ospSetPixelOp(_frameBuffer, _pixelOp);
     ospCommit(_frameBuffer);
+
+    _recreateSubsamplingBuffer();
+
     clear();
 }
 
+void OSPRayFrameBuffer::_recreateSubsamplingBuffer()
+{
+    ospRelease(_subsamplingFrameBuffer);
+    _subsamplingFrameBuffer = nullptr;
+    const auto subsamplingSize = _subsamplingSize();
+    if (_frameSize != subsamplingSize)
+    {
+        _subsamplingFrameBuffer =
+            ospNewFrameBuffer({int(subsamplingSize.x()),
+                               int(subsamplingSize.y())},
+                              _ospFrameBufferFormat(),
+                              OSP_FB_COLOR | OSP_FB_DEPTH);
+        if (_pixelOp)
+            ospSetPixelOp(_subsamplingFrameBuffer, _pixelOp);
+        ospCommit(_subsamplingFrameBuffer);
+    }
+}
+
 void OSPRayFrameBuffer::clear()
 {
     FrameBuffer::clear();
     size_t attributes = OSP_FB_COLOR | OSP_FB_DEPTH;
+    if (_subsamplingFrameBuffer)
+        ospFrameBufferClear(_subsamplingFrameBuffer, attributes);
```

Comment:
Once created the _subsamplingFrameBuffer is never destroyed, is it?
Do you need to clear both buffers? Why not just _currentFB?

Reply:
`clear()` shall clear everything, also the accumulation buffer in the 'normal' framebuffer. `_currentFB()` at the time when `clear()` is called will only clear the subsampling buffer.